service/history/workflowExecutionUtil.go

```diff
@@ -25,6 +25,74 @@ import (
 	"github.com/uber/cadence/common"
 )
 
+type workflowContext interface {
+	getContext() workflowExecutionContext
+	getMutableState() mutableState
+	reloadMutableState() (mutableState, error)
+	getReleaseFn() releaseWorkflowExecutionFunc
+	getRunID() string
+}
+
+type workflowContextImpl struct {
+	context      workflowExecutionContext
+	mutableState mutableState
+	releaseFn    releaseWorkflowExecutionFunc
+}
+
+type updateWorkflowAction struct {
+	noop           bool
+	createDecision bool
+}
+
+var (
+	emptyUpdateAction             = updateWorkflowAction{}
+	updateWorkflowWithNewDecision = updateWorkflowAction{
+		createDecision: true,
+	}
+	updateWorkflowWithoutDecision = updateWorkflowAction{
+		createDecision: false,
+	}
+)
+
+type updateWorkflowActionFunc func(mutableState mutableState) (updateWorkflowAction, error)
+
+func (w *workflowContextImpl) getContext() workflowExecutionContext {
+	return w.context
+}
+
+func (w *workflowContextImpl) getMutableState() mutableState {
+	return w.mutableState
+}
+
+func (w *workflowContextImpl) reloadMutableState() (mutableState, error) {
+	mutableState, err := w.getContext().loadWorkflowExecution()
+	if err != nil {
+		return nil, err
+	}
+	w.mutableState = mutableState
+	return mutableState, nil
+}
```

Comment:
why do we need such function?

Reply:
can we delete this function in the next PR?

Reply:
In update functions UpdateHistoryLoop, each iteration requires a `loadWorkflowExecution` to  get new mutableState.  That's why I think we need this reload. 
Let me know if that's something we can also get rid of.